boards/xtensa/esp32s3_devkitm/esp32s3_devkitm.dts

```diff
@@ -93,6 +93,13 @@
 	pinctrl-names = "default";
 };
 
+&twai {
+	/* requires external CAN transceiver or jumper on RX and TX pins for loopback testing */
```

Comment:
You should be able to use the internal loopback support added in e229898caf23809810a7aff7fbc217f143c0e878 in an overlay where needed. This should void the need for an external loopback jumper.

Reply:
(not a blocker, by the way)

Reply:
I had some trouble getting this working.  I am testing with two different ESP32-S3 boards -- one with a CAN transceiver and one without.  Here are some test results of trying different combinations.

Could you comment on case 2 to see if I used the input-on-output correctly?

### Case 1 - Standard Pinctrl With Separate TX/RX pins

- Pass - ESP32-S3 with CAN Transceiver
- Fail (blocks in `test_filters_added_while_stopped`)  - ESP32-S3 without CAN Transceiver
- Pass - ESP32-S3 without CAN Transceiver and jumper from GPIO13 to GPIO14

Here is the pinctrl DT:
`&pinctrl {
	twai_default: twai_default {
		group1 {
			pinmux = <TWAI_TX_GPIO13>;
		};
                group2 {
			pinmux = <TWAI_RX_GPIO14>;
		};`

### Case 2 - Pinctrl With Input on Output Enabled
Without a CAN transceiver, if I make use of [https://github.com/zephyrproject-rtos/zephyr/commit/e229898caf23809810a7aff7fbc217f143c0e878 e229898], I only get a 52.51% pass rate.  I'm not sure what the issue is here at the moment.  I'm happy to put a scope on anything that you want me to look at.

- Pass - ESP32-S3 with CAN Transceiver
- Fail (blocks in `test_filters_added_while_stopped`)  - ESP32-S3 without CAN Transceiver
- Pass - ESP32-S3 without CAN Transceiver and jumper from GPIO13 to GPIO14

Here is the pinctrl DT:
`&pinctrl {
	twai_default: twai_default {
		group1 {
			pinmux = <TWAI_TX_GPIO13>;
                        input-enable;
		};
                group2 {
			pinmux = <TWAI_RX_GPIO14>;
                        output-enable;
		};`

### Case 3 - Pinctrl With Input on Output Enabled and ESP32 GPIO MUX Signals Crossed
This works and replaces an actual wire jumper.  It is ESP32 specific, but so is the TWAI, so this should be okay.
- Pass - ESP32-S3 with CAN Transceiver
- Pass - ESP32-S3 without CAN Transceiver
- Pass - ESP32-S3 without CAN Transceiver and jumper from GPIO13 to GPIO14

`
&pinctrl {
	twai_default: twai_default {
		group1 {
                        /* Create a virtual jumper connecting TWAI_TX and TWAI_RX */
			pinmux = <ESP32_PINMUX(13, ESP_TWAI_RX, ESP_TWAI_TX)>;
                };
	};
};`

Reply:
Marking this as resolved now that the proper loopback configuration has been updated per Lucas' comments in comment https://github.com/zephyrproject-rtos/zephyr/pull/59918#discussion_r1255875533